src/files.rs

```diff
@@ -44,38 +38,33 @@ impl Locatable for File {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::io::{Seek, SeekFrom, Write, Read};
+    use std::io::{Seek, SeekFrom, Write};
     use tempfile::NamedTempFile;
-    use tempdir::TempDir;
+    //use tempdir::TempDir;
 
-    fn dummy_file() -> (PathBuf, NamedTempFile) {
+    fn dummy_file() -> NamedTempFile {
         let mut temp = NamedTempFile::new().unwrap();
 
         write!(temp, "Hello World!").unwrap();
         temp.seek(SeekFrom::Start(0)).unwrap();
 
-        (PathBuf::from(temp.path()), temp)
+        temp
     }
 
     #[test]
     fn new_file() {
-        let (path, mut f) = dummy_file();
-
-        let file = include_file(&path).unwrap();
+        let f = dummy_file();
 
-        let mut file_contents = Vec::new();
-        f.read_to_end(&mut file_contents).unwrap();
+        let file = include_file(&f.path()).unwrap();
 
-        let mut got = Vec::new();
-        file.contents().unwrap().read_to_end(&mut got).unwrap();
-        assert_eq!(got, file_contents);
-        assert_eq!(file.name(), path);
+        assert_eq!(file.name(), f.path());
+        f.close().unwrap();
     }
 
     #[test]
     fn file_only_works_on_files() {
-        let t = TempDir::new("blah").unwrap();
```

Comment:
I know it's a massive edge case and I doubt anyone would ever have a file called `blah` in their working directory, but do think it's worth inserting an `assert!(!t.exists())`?

Reply:
I never understood what the code was trying to achieve with creating a TempDir. Now you mention that and I realize that I should not have removed this.

Reply:
I think I misread the test when writing that previous comment. The test creates a temporary directory and then makes sure that executing `include_file()` with a directory is an error. So it looks like you might want to put that `TempDir` back in. 

If you can think of a more descriptive name for the test feel free to change it. It may help future developers understand what it's doing.